src/pages/Admin/Room/Edit/RoomEditModal.tsx

```diff
@@ -2,30 +2,40 @@ import React, { useEffect, useState, useCallback, useMemo } from "react";
 import { Form, Modal, Button } from "react-bootstrap";
 import { useForm } from "react-hook-form";
 import ImageInput from "components/atoms/ImageInput";
+import { RoomData_v2 } from "types/rooms";
 
 import { RoomTemplate, ROOM_TEMPLATES } from "settings";
-import { RoomEditModalProps } from "./RoomEditModal.types";
 
 import "./RoomEditModal.scss";
 
+interface RoomEditModalProps {
+  isVisible: boolean;
+  onClickOutsideHandler: () => void;
+  room: RoomData_v2;
+  submitHandler: (values: RoomData_v2, index: number) => void;
+  deleteHandler: () => void;
+}
+
 export const RoomEditModal: React.FC<RoomEditModalProps> = ({
   isVisible,
   onClickOutsideHandler,
   room,
   submitHandler,
   deleteHandler,
 }) => {
-  const [selectedTemplate, setSelectedTemplate] = useState<string>(
-    room.template!
+  const [selectedTemplate, setSelectedTemplate] = useState<string | undefined>(
+    room.template
   );
```

Comment:
I know I originally said to add the `| undefined` here, but apparently `useState` does that for us by default! So we can actually simplify this to just:

```suggestion
  const [selectedTemplate, setSelectedTemplate] = useState<string>(
    room.template
  );
```

Ref:

![image](https://user-images.githubusercontent.com/753891/110567435-a0a9c680-81a5-11eb-837f-a20da6614291.png)

Reply:
I implemented these changes, and at first it was fine, but after resolving conflicts it doesn't work unless I put the `| undefined` in there (same goes for the other useState).

This is the error:
```node
TypeScript error in /Users/vinayramani/Documents/GitHub/sparkle/src/pages/Admin/Room/Edit/RoomEditModal.tsx(28,5):
Argument of type 'string | undefined' is not assignable to parameter of type 'string | (() => string)'.
  Type 'undefined' is not assignable to type 'string | (() => string)'.  TS23451
```

Reply:
What is on `src/pages/Admin/Room/Edit/RoomEditModal.tsx(28,5)` (line 28, column 5) in your branch here?

Looking at the diff, it would seem to be `room.template`:

- https://github.com/sparkletown/sparkle/pull/1193/files#diff-7e274eb670b2bc8208899a93075c06d4f3df2d06bc5964ecf95cd2c20438c66fR28

So what is the type of `room.template`?

Reply:
Actually, disregard the above, looking at both of the overloads for `useState`:

![image](https://user-images.githubusercontent.com/753891/110729156-4a568980-8272-11eb-811c-f29adcf7523b.png)

The `| undefined` part only exists on the 'overload' when we don't pass any default state to it. But because we pass it `room.template`, it will use the other form, which requires `initialState: S | (() => S)`.

So basically, we need to keep it as `useState<string | undefined>`; I was wrong  with my original suggestion.